File: domain_age.js
const { execSync } = require('child_process');
const { DateTime } = require('luxon');

function getDomainAge(domain) {
    try {
        const whoisOutput = execSync(`whois ${domain}`, { encoding: 'utf-8' });
        const creationDateMatch = whoisOutput.match(/Creation Date: (.+)/i);
        if (!creationDateMatch) {
            console.error(`Error: Creation date not found for domain ${domain}`);
            process.exit(1);
        }
        const creationDate = creationDateMatch[1].split("T")[0];
        const age = getRemainingTime(creationDate);
        console.log(`${domain} Created on: ${creationDate}\nAge of ${domain}: ${age}`);
    } catch (error) {
        console.error(`Error: Domain not found or whois server not available for ${domain}`);
        process.exit(1);
    }
}

function getRemainingTime(inputDate) {
    const inputDateObj = DateTime.fromISO(inputDate, { zone: 'utc' });
    const currentDate = DateTime.utc();
    const diffInDays = currentDate.diff(inputDateObj, 'days').days;
    const years = Math.floor(diffInDays / 365);
    const remainingDays = diffInDays % 365;
    const months = Math.floor(remainingDays / 30);
    const remainingDaysOfMonth = Math.floor(remainingDays % 30);
    return `${years} years, ${months} months, and ${remainingDaysOfMonth} days`;
}

if (process.argv.length !== 3) {
    console.error("Usage: node script.js domain_name");
    process.exit(1);
}

const domain = process.argv[2];
getDomainAge(domain);

File: DomainAge.py
import subprocess
from datetime import datetime

def get_domain_age(domain):
    try:
        whois_output = subprocess.check_output(['whois', domain], stderr=subprocess.STDOUT, text=True)
    except subprocess.CalledProcessError as e:
        print(f"Error: Domain not found or whois server not available for {domain}")
        exit(1)
    
    creation_date = None
    for line in whois_output.split('\n'):
        if "Creation Date" in line:
            creation_date = line.split(":")[1].strip().split("T")[0]
            break

    if creation_date:
        age = get_remaining_time(creation_date)
        print(f"{domain} Created on: {creation_date}\nAge of {domain}: {age}")
    else:
        print(f"Creation date not found for domain: {domain}")

def get_remaining_time(input_date):
    input_date_obj = datetime.fromisoformat(input_date.split('T')[0])
    current_date = datetime.now()
    delta = current_date - input_date_obj
    years = delta.days // 365
    remaining_days = delta.days % 365
    months = remaining_days // 30
    remaining_days %= 30
    return f"{years} years, {months} months, and {remaining_days} days"

if __name__ == "__main__":
    import sys
    if len(sys.argv) != 2:
        print("Usage: python script.py domain_name")
        exit(1)
    
    domain = sys.argv[1]
    get_domain_age(domain)

